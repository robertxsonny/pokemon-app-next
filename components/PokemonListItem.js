import Image from 'next/image';
import Link from 'next/link';
import styled from '@emotion/styled';

import { xs } from '../styles/breakpoints';

const TextWrapper = styled.div({
  background: 'linear-gradient(rgba(0,0,0,0) 10%, rgba(0,0,0,0.45) 90%)',
  position: 'absolute',
  bottom: 0,
  left: 0,
  right: 0,
  height: 50,
  transition: '0.3s all ease',
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'flex-end',
  paddingBottom: 12
})

const Name = styled.b({
  color: 'white',
  fontSize: '1rem',
  textTransform: 'capitalize'
})

const Card = styled.a({
  display: 'block',
  backgroundColor: 'lightgrey',
  boxShadow: '0 4px 4px rgba(0, 0, 0, 0.034), 0 6px 6px rgba(0, 0, 0, 0.048)',
  borderRadius: 6,
  width: '100%',
  position: 'relative',
  transition: '0.3s all ease',
  overflow: 'hidden',
  height: 180,
  cursor: 'pointer',
  '&:hover': {
    boxShadow: '0 8px 8px rgba(0, 0, 0, 0.034), 0 12px 12px rgba(0, 0, 0, 0.048)',

    [TextWrapper]: {
      height: 80,
      paddingBottom: 20
    }
  },
  [xs.max]: {
    height: 125
  }
})

const PokemonListItem = ({ name, image, detailUrl, onDelete }) => (
  <Link href={detailUrl}>
    <Card>
      {onDelete && (
        <button onClick={onDelete}>x</button>
      )}
      {image && <Image src={image} layout="fill" objectFit="contain" />}
      <TextWrapper>
        <Name>{name}</Name>
      </TextWrapper>
    </Card>
  </Link>
)

export default PokemonListItem;
