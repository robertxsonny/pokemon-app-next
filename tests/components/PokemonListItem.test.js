import { render } from '@testing-library/react';
import PokemonListItem from "../../components/PokemonListItem";

jest.mock('next/image', () => ({ src, alt }) => <img src={src} alt={alt} />)

describe('PokemonListItem component', () => {
  test('Should render list item component', () => {
    const { container } = render(<PokemonListItem name="Pikachu" detailUrl="/pikachu" />);
    expect(container).toMatchSnapshot();
  })

  test('Should render list item component with subtitle', () => {
    const { container } = render(<PokemonListItem name="Pikachu" subtitle="Pikachu" detailUrl="/pikachu" />);
    expect(container).toMatchSnapshot();
  })

  test('Should render list item component with image', () => {
    const { container } = render(<PokemonListItem name="Pikachu" detailUrl="/pikachu" image="https://images.example.com/pikachu.png" />);
    expect(container).toMatchSnapshot();
  })
})