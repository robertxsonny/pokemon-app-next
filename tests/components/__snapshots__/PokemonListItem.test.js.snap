// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonListItem component Should render list item component 1`] = `
<div>
  <a
    class="css-mrowqa-Card e19yqeoj0"
    href="/pikachu"
  >
    <div
      class="css-1msahx6-TextWrapper e19yqeoj3"
    >
      <b
        class="css-1x9t9rc-Name e19yqeoj2"
      >
        Pikachu
      </b>
    </div>
  </a>
</div>
`;

exports[`PokemonListItem component Should render list item component with image 1`] = `
<div>
  <a
    class="css-mrowqa-Card e19yqeoj0"
    href="/pikachu"
  >
    <img
      alt="Pikachu"
      src="https://images.example.com/pikachu.png"
    />
    <div
      class="css-1msahx6-TextWrapper e19yqeoj3"
    >
      <b
        class="css-1x9t9rc-Name e19yqeoj2"
      >
        Pikachu
      </b>
    </div>
  </a>
</div>
`;

exports[`PokemonListItem component Should render list item component with subtitle 1`] = `
<div>
  <a
    class="css-mrowqa-Card e19yqeoj0"
    href="/pikachu"
  >
    <div
      class="css-1msahx6-TextWrapper e19yqeoj3"
    >
      <b
        class="css-1x9t9rc-Name e19yqeoj2"
      >
        Pikachu
      </b>
      <span
        class="css-1f5ct2u-Subtitle e19yqeoj1"
      >
        Pikachu
      </span>
    </div>
  </a>
</div>
`;
